apps/admin/src/pages/form/utils/filterFormResponse.ts

```diff
@@ -1,36 +1,76 @@
-const allowLists: Record<string, string[]> = {
-  panel: ['label', 'key', 'type', 'input', 'components'],
-  radio: ['label', 'description', 'key', 'type', 'input', 'values'],
-  select: ['label', 'description', 'key', 'type', 'input', 'widget', 'placeholder', 'data'],
-  selectboxes: ['label', 'description', 'key', 'type', 'input', 'values'],
-  textarea: ['label', 'description', 'key', 'type', 'input', 'autoExpand', 'maxCharCount'],
-  textfield: ['label', 'description', 'key', 'type', 'input'],
+import {
+  $FormCheckboxComponentInput,
+  $FormComponentInputValidate,
+  $FormInput,
+  $FormPanelComponentInput,
+  $FormRadioComponentInput,
+  $FormSelectComponentInput,
+  $FormTextAreaComponentInput,
+  $FormTextFieldComponentInput,
+} from '@meldingen/api-client'
+import filter from 'uber-json-schema-filter'
+
+const filterBySchemaPerType = (obj: any) => {
+  switch (obj.type) {
+    case 'panel':
+      return filter($FormPanelComponentInput, obj)
+    case 'radio':
+      return filter($FormRadioComponentInput, obj)
+    case 'select':
+      return filter($FormSelectComponentInput, obj)
+    case 'selectboxes':
+      return filter($FormCheckboxComponentInput, obj)
+    case 'textarea':
+      // Add autoExpand to object if it doesn't exist, the builder doesn't do that by default
+      if (!obj.hasOwnProperty('autoExpand')) {
+        return filter($FormTextAreaComponentInput, { ...obj, autoExpand: false })
+      }
+
+      return filter($FormTextAreaComponentInput, obj)
+    case 'textfield':
+      return filter($FormTextFieldComponentInput, obj)
+    default:
+      throw Error(`Type ${obj.type} is unknown, please add it to filterFormResponse.`)
+  }
+}
+
+const getFilteredValidateObject = (validateObj: any) => {
+  const validate = filter($FormComponentInputValidate, validateObj)
+
+  // Explicitly remove the 'json' key if its value is an empty string, the API doesn't accept that
+  if (validate?.json === '') {
+    delete validate.json
+  }
+
+  return validate
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const filterFormResponse = (obj: any): any => {
-  // If the input is a JS primitive, return the unmodified input
-  if (obj !== Object(obj)) {
-    return obj
-  }
+  const firstLevelComponents = obj.components.map((firstLevelComponent: any) => {
+    const secondLevelComponents = firstLevelComponent.components.map((secondLevelComponent: any) => {
+      return {
+        ...filterBySchemaPerType(secondLevelComponent),
+        validate: getFilteredValidateObject(secondLevelComponent.validate),
+      }
+    })
```

Comment:
Maybe a comment for firstLevelComponents and secondLevelComponents. It suggests some fixed object schema for the input, but you cannot refer to it with any

Reply:
What do you mean with 'you cannot refer to it with any'? 

In the builder, every questionnaire is a wizard with panels, which in turn contain questions. So in that sense, the schema has a fixed maximum depth of 2 (we could change that if we'd allow separate 'panel' components, but we probably won't).

With that information, could you suggest a comment? Or, preferably, a better name for first- and secondLevelComponents? I'm not sure what you're looking for.

Reply:
Something like:
A questionnaire can have a wizard with panels containing questions. Therefore the object has a fixed depth of two levels.